openwisp_notifications/api/serializers.py

```diff
@@ -72,7 +73,7 @@ class Meta(NotificationSerializer.Meta):
         list_serializer_class = CustomListSerializer
 
 
-class NotificationSettingSerializer(serializers.ModelSerializer):
+class NotificationSettingSerializer(ValidatedModelSerializer):
```

Comment:
The serializer was not using `ValidatedModelSerializer` earlier. Thus, it was not calling the model's full_clean method. Due to this, the database was not storing `None` value. 

Thanks for catching this.